scriptlets/sru-helpers/fetch_manifest

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+#
+# Copyright 2024 Canonical Ltd.
+#
+# Fetch the manifest for a given machine from C3
+#
+# Usage:
+#     `fetch_manifest <CID> <MACHINE_SECURE_ID>`
+
+from argparse import ArgumentParser
+from io import StringIO
+import requests
+
+
+# This is the manifest that all machines will use if we cannot find
```

Comment:
If the purpose of this script is to retrieve a manifest using the API, then I don't think the script itself is the best place to define the default manifest. Perhaps it should be in a separate file but my suggestion would be for the API itself to return the default manifest.

Reply:
It would still be useful if the endpoint (assuming it's reachable) returned a default manifest if the requested one wasn't located. Or perhaps there could be a variation of the endpoint that you would hit just to retrieve a default manifest, in case the original request was unsuccessful.

Reply:
I would also argue that this script shouldn't return a hardwired default manifest in any case. It's purpose is to retrieve a manifest from the endpoint. If the endpoint is unreachable then the script exits with an error code because it has been unsuccessful in fulfilling its purpose.

Reply:
I have updated the script to exit with exit code 1 in case it cannot fetch the manifest. The user of the script can then from the exit code whether the manifest is fetched or not and use the default instead.